lib/awful/mouse/init.lua

```diff
@@ -280,6 +280,32 @@ function mouse.resize_handler(c, context, hints)
     end
 end
 
+--- Get the client currently under the mouse cursor.
+-- @property current_client
+-- @tparam client|nil The client
+
+function mouse.object.get_current_client()
+    local obj = capi.mouse.object_under_pointer()
+    if type(obj) == "client" then
+        return obj
+    end
+end
+
+function mouse.object.set_current_client() end
+
+--- Get the wibox currently under the mouse cursor.
+-- @property current_wibox
+-- @tparam wibox|nil The wibox
+
+function mouse.object.get_current_wibox()
+    local obj = capi.mouse.object_under_pointer()
+    if type(obj) == "drawin" then
+        return obj
+    end
+end
```

Comment:
Sorry, but a drawin is not a wibox. For example, try calling `get_current_wibox():setup()` or `:set_widget()`. This really returns a drawin and thus this function is useless in 99% of cases.

Can't we just deprecate it without a replacement? :-(
If not, then this has to be called `get_current_drawin()` and needs a warning about a drawin not being a wibox etc.